skrifa/src/meta/strings.rs

```diff
@@ -0,0 +1,683 @@
+//! Strings describing font names and other metadata in multiple languages.
+//!
+//! This provides higher level interfaces for accessing the data in the
+//! OpenType [name](https://learn.microsoft.com/en-us/typography/opentype/spec/name)
+//! table.
+//!
+//! # Example
+//! The following function will print all localized strings in a font:
+//! ```
+//! use skrifa::MetadataProvider;
+//!
+//! fn print_all_strings<'a>(font: &impl MetadataProvider<'a>) {
+//!     for id in font.string_ids() {
+//!         println!("[{:?}]", id);
+//!         for string in font.localized_strings(id) {
+//!             println!("{:?} {}", string.language(), string.to_string());
+//!         }
+//!     }
+//! }
+//! ```
+
+use read_fonts::{
+    tables::name::{CharIter, Name, NameRecord, NameString},
+    TableProvider,
+};
+
+use core::fmt;
+
+#[doc(inline)]
+pub use read_fonts::types::NameId as StringId;
+
+#[derive(Clone)]
+/// Iterator over the characters of a string.
+pub struct Chars<'a> {
+    inner: Option<CharIter<'a>>,
+}
+
+impl<'a> Iterator for Chars<'a> {
+    type Item = char;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.as_mut()?.next()
+    }
+}
+
+/// Iterator over the collection of available string identifiers in a font.
+pub struct StringIds<'a> {
+    records: core::slice::Iter<'a, NameRecord>,
+    last_id: Option<StringId>,
+}
+
+impl<'a> StringIds<'a> {
+    /// Creates a new string identifier iterator from the given font.
+    pub fn new(font: &impl TableProvider<'a>) -> Self {
+        let name = font.name().ok();
+        let records = name
+            .as_ref()
+            .map(|name| name.name_record().iter())
+            .unwrap_or([].iter());
+        Self {
+            records,
+            last_id: None,
+        }
+    }
+}
+
+impl<'a> Iterator for StringIds<'a> {
+    type Item = StringId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let record = self.records.next()?;
+            let id = Some(record.name_id());
+            if self.last_id < id {
+                self.last_id = id;
+                return id;
+            }
+        }
+    }
+}
+
+/// Iterator over a collection of localized strings for a specific identifier.
+#[derive(Clone)]
+pub struct LocalizedStrings<'a> {
+    name: Option<Name<'a>>,
```

Comment:
again I find it slightly odd that we will create this iterator even if there is no `Name` table?